### ISSUE TITLE
'Implicitly unwrapped optionals' rule should not flag Apple APIs

### ISSUE THREAD

**Issue by BenStaveleyTaylor** [realm/SwiftLint#2271]
The implicitly_unwrapped_optional rule generates warnings on Apple-defined delegate methods that I can't change. For example, in WKNavigationDelegate there are many `WKNavigation!` parameters due to them being `null_unspecified` in the Objective C headers.

The rule is smart enough to not flag implicitly unwrapped optionals when used as nib outlets. Is there a way to ignore this situation too?

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --config swiftlint.yml 
Loading configuration from 'swiftlint.yml'
Linting Swift files in current working directory
Linting 'Test.swift' (1/1)
/Users/bestave/Desktop/SwiftLint issue/Test.swift:5:47: warning: Implicitly Unwrapped Optional Violation: Implicitly unwrapped optionals should be avoided when possible. (implicitly_unwrapped_optional)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
-- 0.26.0

* Installation method used (Homebrew, CocoaPods, building from source, etc)?
--  Installed SwiftLint.pkg

* Paste your configuration file:

```yml
opt_in_rules:
- implicitly_unwrapped_optional
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
-- No

* Which Xcode version are you using (check `xcode-select -p`)?
-- Version 9.4.1 (9F2000)

* Do you have a sample that shows the issue? 
-- Yes:
```swift
import WebKit

public class Test: WKNavigationDelegate {

    public func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
        return
    }

}
```

**Comment by marcelofabri** [comment]
Currently SwiftLint doesn't know if the method you're implementing is from a protocol. We could have a configuration that would allow implicitly unwrapped optionals on function parameters.

However, since this is an opt-in rule, false positives are expected.
